File: juomapeli.js
var players = ['Jere','Kati'];
var number_of_rounds = 50;
var max_difference_between_players_amount = 5;
var max_amount_of_drinks_at_once = 5;
var play = 0;
var players_amount = Array.apply(null, new Array(players.length)).map(Number.prototype.valueOf,0);
var empty = [0]; 

  function updatelist()
  {
    $('#pelaajat').html('');
    for(var i = 0;i < players.length;i++)
    {
      var pelaaja_teksti = "<a>" + players[i] +" "+ players_amount[i]+ " " + "</a>";
      $("#pelaajat").append(pelaaja_teksti);
    }
  }
    //Uusi pelaaja
    function newplayer()
    {
      document.getElementById('id01').style.display='none'
      var nameValue = document.getElementById("uuspelaaja").value;
      if (nameValue != '')
      {
        players.push(nameValue);
        players_amount.push(0);
        updatelist();
      }
    }
    $(document).keypress(
      function(event){
       if (event.which == '13') {
        newplayer();
        event.preventDefault();
      }
    });

    //Puhe
    function speak(teksti)
    {
      responsiveVoice.speak(teksti,'Finnish Female');
      $("#text_line").text(teksti);
    }

    //juomafunktiot
    function drink(player)
    {
      var amount = Math.floor((Math.random() * max_amount_of_drinks_at_once) + 1);
      players_amount[player] = players_amount[player] + amount;
      line = players[player] + " juo " + amount.toString();
      speak(line);
      
    }

    function all_drink()
    {
    	var amount = Math.floor((Math.random() * max_amount_of_drinks_at_once) + 1);
    	for(var i = 0;i < players_amount;i++)
      {
        players_amount[i] = players_amount[i] + amount;
      }
      line = 'Kaikki juo ' + amount.toString();
      speak(line);
    }
    //Kuka juo vai kaikki juo
    function round()
    {
      var jackpot = Math.floor((Math.random() * 15) + 1);
      //all drink
      if (jackpot == 1){all_drink();}
      //one drink
      else
      {
        if (Math.max.apply(Math, players_amount) - Math.min.apply(Math, players_amount) > max_difference_between_players_amount)
        {
          player = players_amount.indexOf(Math.min.apply(Math, players_amount));
          
        }
        else
        {
         player = Math.floor((Math.random() * players.length) + 0);
         
       }
       drink(player);
     }
     updatelist();
   }

   function playGame()
   {
    speak("Juomapeli alkaa");

    
    var time_between_drinks = 5;

    for(var rounds = 0;rounds < number_of_rounds;rounds++)
    {
        empty.push(setTimeout(round, time_between_drinks*1000));
        time_between_drinks += Math.floor((Math.random() * 20) + 20);
    }

  }
  function stopGame(){
    for(var rounds = 0;rounds < number_of_rounds;rounds++)
    {
      clearTimeout(empty[rounds]);
    } 
    speak('Peli päättyi');
    players_amount = Array.apply(null, new Array(players.length)).map(Number.prototype.valueOf,0);
  }



  $(document).ready(function(){
    updatelist();
    $("#text_line").text("Kivoja tupareita <3");
  });

File: py/juomapeli.py
from os import system
import time
import random
import platform

players = ['Johannes','Skeletor','Muikku']
number_of_rounds = 50
max_difference_between_players_amount = 5
max_amount_of_drinks_at_once = 5
time_between_drinks = random.randint(15,30)

players_amount = [0] * len(players)

def speak(string):
	
	system("""osascript -e 'tell application "Spotify" to set sound volume to 70'""")
	time.sleep(0.1)
	system("""osascript -e 'tell application "Spotify" to set sound volume to 50'""")
	time.sleep(0.1)
	system("""osascript -e 'tell application "Spotify" to set sound volume to 35'""")
	time.sleep(0.2)
	system('say ' + string)
	time.sleep(0.2)
	system("""osascript -e 'tell application "Spotify" to set sound volume to 50'""")
	time.sleep(0.1)
	system("""osascript -e 'tell application "Spotify" to set sound volume to 90'""")
	time.sleep(0.1)
	system("""osascript -e 'tell application "Spotify" to set sound volume to 100'""")


def drink(player):
	amount = random.randint(1,max_amount_of_drinks_at_once)
	players_amount[player] = players_amount[player] + amount
	line = players[player] + ' juo ' + str(amount)
	speak(line)

def all_drink():
	amount = random.randint(1,max_amount_of_drinks_at_once)
	for i in range(0,len(players_amount)):
		players_amount[i] = players_amount[i] + amount
	line = 'Kaikki juo ' + str(amount)
	speak(line)

def results():
	speak('Tulokset')
	time.sleep(0.5)
	for player in range(0,len(players)):
		speak(players[player] + ' joi ' + str(players_amount[player]))
		time.sleep(0.1)
	speak('Paras suorittaja oli' + players[players_amount.index(max(players_amount))])
	speak('Huonoin suorittaja oli' + players[players_amount.index(min(players_amount))])

def main():
	system("""osascript -e 'tell application "Spotify" to set sound volume to 100'""")
	speak('Juomapeli alkaa')
	time.sleep(5)

	for rounds in range(1,number_of_rounds):

		jackpot = random.randint(1,15)
		if (jackpot == 1):
			all_drink()
		else:
			if (max(players_amount) - min(players_amount) > max_difference_between_players_amount):
				player = players_amount.index(min(players_amount))
			else:
				player = random.randint(0,len(players)-1)

			drink(player)

		print('Kierros: '+ str(rounds))
		print(players)
		print(players_amount)
		time_between_drinks = random.randint(15,30)
		time.sleep(time_between_drinks)

	speak('Juomapeli loppui')
	results()

main()
